src/components/icon-button/IconButton.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+
+import Icon from '../icon';
+
+import {
+    DEFAULT_ICON_BUTTON_ELEMENT,
+    DEFAULT_ICON_BUTTON_SIZE,
+    ICON_BUTTON_DEFINITIONS,
+} from './IconButton.constants';
+import PIconButton from './IconButton.props';
+import IconButtonRoot from './IconButton.root';
+
+const IconButton = ({
+    element = DEFAULT_ICON_BUTTON_ELEMENT,
+    size = DEFAULT_ICON_BUTTON_SIZE,
+    icon,
+    label,
+    ...rest
+}: PIconButton): JSX.Element => (
+    <IconButtonRoot
+        as={element}
```

Comment:
do we need to pass in another element? I like the implementation that this is a styled button only and it might help with semantics to keep it that way

Reply:
This might be worth discussing, but frankly speaking you are right here! 👍🏻

Reply:
On second thought ... if we want to keep it generic (for implementations outside MM) we should leave it like that, even if we do not use this functionality ourselves.